front/components/assistant_builder/InstructionScreen.tsx

```diff
@@ -0,0 +1,366 @@
+                {CREATIVITY_LEVELS.map(({ label, value }) => (
+                  <DropdownMenu.Item
+                    key={label}
+                    label={label}
+                    onClick={() => {
+                      setGenerationSettings({
+                        ...generationSettings,
+                        temperature: value,
+                      });
+                    }}
+                  />
+                ))}
+              </DropdownMenu.Items>
+            </DropdownMenu>
+          </div>
+        </div>
+      </DropdownMenu.Items>
+    </DropdownMenu>
+  );
+}
+
+function AssistantBuilderTextArea({
+  placeholder,
+  value,
+  onChange,
+  error,
+  name,
+}: {
+  placeholder: string;
+  value: string | null;
+  onChange: (value: string) => void;
+  error?: string | null;
+  name: string;
+}) {
+  return (
+    <textarea
+      name="name"
+      id={name}
+      className={classNames(
+        "block max-h-full min-h-48 w-full min-w-0 grow rounded-md text-sm text-sm",
+        !error
+          ? "border-gray-300 focus:border-action-500 focus:ring-action-500"
+          : "border-red-500 focus:border-red-500 focus:ring-red-500",
+        "bg-structure-50 stroke-structure-50",
+        "resize-none"
+      )}
+      placeholder={placeholder}
+      value={value ?? ""}
+      onChange={(e) => {
+        onChange(e.target.value);
+      }}
+    />
+  );
+}
+
+const STATIC_SUGGESTIONS = [
+  "I want you to act as the king of the bongo.",
+  "I want you to act as the king of the bongo, Bong.",
+  "I want you to act as the king of the cats, Soupinou.",
+];
+
+const SUGGESTION_DEBOUNCE_DELAY = 1500;
+
+function Suggestions({ instructions }: { instructions: string | null }) {
+  const [suggestions, setSuggestions] = useState<string[]>(STATIC_SUGGESTIONS);
+  const [loading, setLoading] = useState(false);
+  const debounceHandle = useRef<NodeJS.Timeout | undefined>(undefined);
+  useEffect(() => {
+    if (debounceHandle.current) {
+      clearTimeout(debounceHandle.current);
+      debounceHandle.current = undefined;
+    }
+    if (!instructions) {
+      setSuggestions(STATIC_SUGGESTIONS);
+    }
+    if (instructions) {
+      // Debounced request to generate suggestions
+      debounceHandle.current = setTimeout(async () => {
+        setLoading(true);
+        const suggestions = await getInstructionsSuggestions(instructions);
+        setSuggestions(suggestions);
+        setLoading(false);
+      }, SUGGESTION_DEBOUNCE_DELAY);
+    }
+  }, [instructions]);
+
+  return (
+    <Collapsible defaultOpen>
+      <div className="flex flex-col gap-2">
+        <Collapsible.Button>
+          <div className="text-base font-bold text-element-800">
+            Suggestions
+          </div>
+        </Collapsible.Button>
+        <Collapsible.Panel>
+          <div className="flex gap-2">
+            {loading && <Spinner size="sm" />}
+            {!loading &&
+              suggestions.map((suggestion, index) => (
```

Comment:
NIT: Should we .slice(0, 3) to ensure we have only max 3 results?

Reply:
But I told the LLM not to output more than 3... so yeah, definitely slice :laughing:

Reply:
I did the fix in the following PR #3974 (in which you're pinged too, and in which the risk of having more than 3 actually materializes)